library/cc/bridge_utility.cc

```diff
@@ -0,0 +1,70 @@
+#include "bridge_utility.h"
+
+#include <sstream>
+
+namespace Envoy {
+namespace Platform {
+
+// TODO(crockeo): we always copy memory across boundaries; consider allowing for moves and/or
+// shared ownership w/ reference counting via envoy-mobile's release callbacks
+
+envoy_data string_as_envoy_data(const std::string& str) {
+  uint8_t* bytes = static_cast<uint8_t*>(safe_malloc(str.size()));
+  memcpy(bytes, &str[0], str.size());
+  return envoy_data{
+      .length = str.size(),
+      .bytes = bytes,
+      .release = envoy_noop_release,
```

Comment:
`release` should probably be `free` here (since you're using `malloc`).

Reply:
ohhh my bad. i should have looked at the implementation of `envoy_noop_release`; i assumed that it was a `free` call w/o further action (noop). in retrospect it's obviously `noop` = no free :grin: